students/matt_casali/Lesson09/mailroom_9.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+import datetime, sys
+now = datetime.datetime.now()
+
+class Donor:
+    def __init__(self, first_name, last_name, donation_amount):
+        self.first = first_name
+        self.last = last_name
+        self.donation = donation_amount
+
+    @property
+    def full_name(self):
+        return f"{self.first} {self.last}"
+
+    def add_donation(self, amount):
+        return self.donation.append(amount)
+
+    def sum_donation(self):
+        return sum(self.donation)
+
+
+def get_name():
+    return input("Who would you like to thank? If you would like a list of donors, enter 'list'. ")
+
+
+def get_amount():
+    try:
+        return float(input(f"How much did this person donate? "))
+    except ValueError:
+        print("You did not enter a numeric value. Please try again.")
+        sys.exit()
+
+
+class Donor_Actions:
```

Comment:
per instructions your data classes should not contain user interactions (calls to `input()`)

your second data class should deal with a collection of Donor objects.

Reply:
Hey Natasha, I made sure to keep the functions involving and calls to input() outside of the two classes I created. Those two functions are called on within the Donor_Actions class. Is there a better way I should approach this problem? Thanks!

Reply:
Even though `input()` isn't called directly - it calls a function that wraps the input function, which ends up being the same.


Your two data classes should be purely dealing with data then your program flow code will interact with these objects.

Reply:
Okay, I think that makes sense. I've added the calls for input() to the main() function so the calls stay outside of the Donor_Actions class. 

I also went back and added more methods for average donation, the number of donations, the text within the thank you letter, and a few others. 

Would it be possible for to resubmit the exercise?

Reply:
Absolutely - resubmit via a new PR and I will review/adjust the grade accordingly.

Reply:
Hey Natasha, just submitted another PR with the new code.

https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1493

Thanks for the help with this!